tests/org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/validation/ValidationBug427432Test.xtend

```diff
@@ -41,7 +41,7 @@ class ValidationBug427432Test extends AbstractXtendTestCase {
 				}
 			
 			}
-		'''.parse.assertError(XbasePackage.Literals.XFEATURE_CALL, IssueCodes.DUPLICATE_CASE)
+		'''.parse.assertNoErrors
```

Comment:
My gut feeling is that the SwitchConstantExpressionsInterpreter needs two modes - one for the compiler and another for the validation. No longer detecting problematic code seems to be a bad side effect of the compiler fix that's worth to reconsider the nature of the fix itself.